### PR TITLE
Add itl http and dns changes

### PR DESCRIPTION
**Pull request by mzac**
Add two options to the ITL DNS and HTTP checks

### CODE REVIEW

File: itl/command-plugins.conf [hunk 1]
```diff
@@ -1532,6 +1536,10 @@ object CheckCommand "dns" {
 			value = "$dns_server$"
 			description = "Optional DNS server you want to use for the lookup."
 		}
+		"-q" = {
+			value = "$dns_query_type$"
+			description = "Optional DNS record query type where TYPE =(A, AAAA, SRV, TXT, MX, ANY). The default query type is 'A' (IPv4 host entry)"
+		}
```
Comment (Al2Klimov) [inline itl/command-plugins.conf:1542]: -q? AFAIK there's no -q:

https://github.com/monitoring-plugins/monitoring-plugins/blob/739a7bbe7ceacab9809998d9bf06a4460b44c522/plugins/check_dns.c#L547-L563

File: itl/command-plugins.conf [hunk 2]
```diff
@@ -367,6 +367,10 @@ object CheckCommand "http" {
 		"--sni" = {
 			set_if = "$http_sni$"
 		}
+		"--verify-host" = {
+			set_if = "$http_verify_host$"
+			description = "Verify SSL certificate is for the -H hostname (with --sni and -S)"
+		}
```
Comment (Al2Klimov) [inline itl/command-plugins.conf:373]: Same here:

https://github.com/monitoring-plugins/monitoring-plugins/blob/739a7bbe7ceacab9809998d9bf06a4460b44c522/plugins/check_http.c

**Comment by mzac** [comment]
@Al2Klimov I am running on a Redhat system.  We seem to have an internal repo but it seems to be mirroring the plugins from www.nagios-plugins.org  

```
Installed Packages
Name        : nagios-plugins-http
Arch        : x86_64
Version     : 2.3.3
Release     : 2.el7
Size        : 106 k
Repo        : installed
From repo   : McGill_University_EPEL_7Server_EPEL_7Server_x86_64
Summary     : Nagios Plugin - check_http
URL         : https://www.nagios-plugins.org/
License     : GPLv2+
Description : Provides check_http support for Nagios.
```

These packages do have these options:
https://github.com/nagios-plugins/nagios-plugins/blob/master/plugins/check_http.c#L238
https://github.com/nagios-plugins/nagios-plugins/blob/master/plugins/check_dns.c#L211

File: doc/10-icinga-template-library.md [hunk 3]
```diff
@@ -322,6 +322,7 @@ Name                 | Description
 ---------------------|--------------
 dns_lookup           | **Optional.** The hostname or IP to query the DNS for. Defaults to "$host_name$".
 dns_server           | **Optional.** The DNS server to query. Defaults to the server configured in the OS.
+dns_query_type       | **Optional.** The DNS record query type where TYPE =(A, AAAA, SRV, TXT, MX, ANY). The default query type is 'A' (IPv4 host entry)
```
Comment (Al2Klimov) [inline doc/10-icinga-template-library.md:325]: IMO "where TYPE =" is redundant.

File: itl/command-plugins.conf [hunk 4]
```diff
@@ -1532,6 +1536,10 @@ object CheckCommand "dns" {
 			value = "$dns_server$"
 			description = "Optional DNS server you want to use for the lookup."
 		}
+		"-q" = {
+			value = "$dns_query_type$"
+			description = "Optional DNS record query type where TYPE =(A, AAAA, SRV, TXT, MX, ANY). The default query type is 'A' (IPv4 host entry)"
```
Comment (Al2Klimov) [inline itl/command-plugins.conf:1541]: Same here.

File: itl/command-plugins.conf [hunk 5]
```diff
@@ -1554,6 +1562,7 @@ object CheckCommand "dns" {
 	}
 
 	vars.dns_lookup = "$host.name$"
+	vars.dns_query_type = "A"
```
Comment (Al2Klimov) [inline itl/command-plugins.conf:1565]: Is a default value actually required?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
1. Drop the http_verify_host commits in favor of #8005.
2. Add [a note](https://github.com/Icinga/icinga2/pull/8005/files#diff-e3021f9a13beebb1e13d11aba1923e31R667) telling dns_query_type is supported only by Nagios plugins.
3. Squash the dns_query_type commits.
3.

**Comment by julianbrost** [comment]
@mzac I'm closing this for now as it hasn't seen any activity for some time now and we don't consider this change important enough to put in the time to fix the remaining issues ourselves. If you still want to work on this, feel free to reopen it at any time.

**Pull request closed without merging**